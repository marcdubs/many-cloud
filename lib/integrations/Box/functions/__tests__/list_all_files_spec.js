describe("list_all_files", () => {
  const expected_resolution = "watup fam";
  let data, func;

  beforeEach(() => {
    var counter = 0;
    data = {
      client: {
        folders: {
          getItems: jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
              if (counter >= 10) {
                resolve([]);
                return;
              }
              resolve([expected_resolution + counter]);
              counter++;
            });
          })
        }
      }
    };
    let list_files = require("../list_files")(data);
    func = require("../list_all_files")(data, list_files);
  });

  describe("when there are no errors", () => {
    let actual;
    beforeEach(done => {
      func("id").then(res_ => {
        actual = res_;
        done();
      });
    });

    it("calls data.client.folders.getItems", () => {
      expect(data.client.folders.getItems).toHaveBeenCalled();
    });

    it("passes in correct default parameters", () => {
      expect(data.client.folders.getItems.mock.calls[0][1]).toMatchSnapshot();
    });

    it("returns expected resolution", () => {
      expect(actual).toMatchSnapshot();
    });
  });

  describe("when getItems throws an error", () => {
    const expected_error = "NOOOOOO";
    beforeEach(() => {
      data.client.folders.getItems.mockImplementation(() => {
        return new Promise((resolve, reject) => {
          throw new Error(expected_error);
        });
      });
    });

    it("rejects with the error", async () => {
      await expect(func("id")).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
