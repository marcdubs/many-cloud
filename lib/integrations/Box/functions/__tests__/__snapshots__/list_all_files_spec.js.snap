// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list_all_files when getItems throws an error rejects with the error 1`] = `"NOOOOOO"`;

exports[`list_all_files when there are no errors passes in correct default parameters 1`] = `
Object {
  "fields": "name",
  "limit": 100,
  "offset": 0,
}
`;

exports[`list_all_files when there are no errors returns expected resolution 1`] = `
Object {
  "entries": Array [
    "watup fam1",
    "watup fam2",
    "watup fam3",
    "watup fam4",
    "watup fam5",
    "watup fam6",
    "watup fam7",
    "watup fam8",
    "watup fam9",
    "watup fam10",
  ],
  "total_count": 10,
}
`;
